File: memory-domain/src/main/java/com/memory/dto/game/response/GameSettingResponse.java
package com.memory.dto.game.response;

import com.memory.domain.game.GameMode;
import com.memory.domain.game.GameSetting;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
public class GameSettingResponse {
    
    private final Long id;
    private final GameMode gameMode;
    private final Integer maxQuestions;
    private final Integer timeLimitSeconds;
    private final Integer maxDistanceForFullScoreKm;
    private final String scoringFormula;
    private final Boolean isActive;
    private final LocalDateTime createDate;
    
    public GameSettingResponse(GameSetting gameSetting) {
        this.id = gameSetting.getId();
        this.gameMode = gameSetting.getGameMode();
        this.maxQuestions = gameSetting.getMaxQuestions();
        this.timeLimitSeconds = gameSetting.getTimeLimitSeconds();
        this.maxDistanceForFullScoreKm = gameSetting.getMaxDistanceForFullScoreKm();
        this.scoringFormula = gameSetting.getScoringFormula();
        this.isActive = gameSetting.getIsActive();
        this.createDate = gameSetting.getCreateDate();
    }
    
    public static GameSettingResponse from(GameSetting gameSetting) {
        return new GameSettingResponse(gameSetting);
    }
}

File: memory-api/src/main/kotlin/com/memory/dto/game/response/GameSettingResponse.kt
package com.memory.dto.game.response

import com.memory.domain.game.GameMode
import com.memory.domain.game.GameSetting
import java.time.LocalDateTime

class GameSettingResponse(gameSetting: GameSetting) {
    private val id: Long?
    private val gameMode: GameMode
    private val maxQuestions: Int
    private val timeLimitSeconds: Int
    private val maxDistanceForFullScoreKm: Int
    private val scoringFormula: String
    private val isActive: Boolean
    private val createDate: LocalDateTime?

    init {
        this.id = gameSetting.id
        this.gameMode = gameSetting.gameMode
        this.maxQuestions = gameSetting.maxQuestions
        this.timeLimitSeconds = gameSetting.timeLimitSeconds
        this.maxDistanceForFullScoreKm = gameSetting.maxDistanceForFullScoreKm
        this.scoringFormula = gameSetting.scoringFormula
        this.isActive = gameSetting.isActive
        this.createDate = gameSetting.createDate
    }

    companion object {
        fun from(gameSetting: GameSetting): GameSettingResponse {
            return GameSettingResponse(gameSetting)
        }
    }
}
